### PR TITLE
added parmoo to staged recipes

### PR DESCRIPTION
**Pull request by thchang**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thchang** [comment]
@conda-forge/help-python I have attempted to add our Python package ``parmoo`` to the staged recipes, by generating the recipe for pypi using grayskull, as instructed on: https://conda-forge.org/docs/maintainer/adding_pkgs.html

I have the following concerns about what I did:
 - How to handle optional requirements?  Currently, parmoo has one optional dependency https://github.com/libensemble/libensemble (also on conda-forge and pypi).  I don't see how to add optional dependencies here (e.g., on PyPI I had an ``extras_requires`` field in the ``setup.py`` file), but grayskull didn't generate anything here.  Similarly, to build the docs one needs ``sphinx``, ``sphinx_rtd_theme``, and ``sphinxcontribut.bibtex``, and to run tests one needs ``pytest``, ``pytest-cov``, and ``flake8``, but I don't see a way to add those either.
 - How to update requirements in the future?  We have another release coming out soon.  When that happens should I open a PR to update the ``version`` field in my yaml file?  Also, the update will add several new dependencies, anything extra I need to do for that?
 - Running tests:  I tried to copy advice from here:  https://github.com/conda-forge/conda-forge.github.io/issues/345 on how to have conda automatically run unit tests using pytest, but I don't know how to check whether I did it correctly.

Many thanks for advice (I am a conda noob),
---Tyler

File: recipes/parmoo/meta.yaml
Comment (dopplershift) [inline recipes/parmoo/meta.yaml:34]: Looks like a test is failing due to missing `mpi4py`:
```suggestion
    - libensemble
    - mpi4py
```

File: recipes/parmoo/meta.yaml
Comment (dopplershift) [inline recipes/parmoo/meta.yaml:20]: Is tox actually necessary to build the package?
```suggestion
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just some minor changes, one of which should help get everything green.

If you want to see the results of the builds yourself, you can click the "Details" link below, and follow the links to the build logs on Azure.

Once this recipe is merged, it will create a new "feedstock", where you'll have the ability to make and merge Pull Requests to maintain the recipe. There, in addition to bumping the version/hash for new releases, you can adjust the dependencies, build steps, etc. See [the docs](https://conda-forge.org/docs/maintainer/updating_pkgs.html) for more info.

Conda doesn't have a concept of optional dependencies, so you can either rely on users to install them manually as needed, or just add them as required dependencies. It depends what makes the most sense for this particular package.

File: recipes/parmoo/meta.yaml
Comment (thchang) [reply recipes/parmoo/meta.yaml]: no, tox is not required.  I missed that.  I think it was mistakenly grabbed by grayskull

Comment (thchang) [reply recipes/parmoo/meta.yaml]: removed it in the latest push

File: recipes/parmoo/meta.yaml
Comment (thchang) [reply recipes/parmoo/meta.yaml]: So this is an optional dependency of libensemble that I am not using for my tests.  It thinks I am trying to use it because libensemble runs cannot be started through pytest, which causes an error.  Instead, I've updated my pytest recipe to only run unit tests (not regression tests), this should solve the problem.  To run the full test script, we'd need to run a special test script that I wrote, contained in the parmoo/tests directory

**Comment by thchang** [comment]
everything appears to be passing now (running unit tests only).  FYI, I have preemptively added a couple of optional dependencies to the requirements list, which are not needed yet but will be needed in the next release (in a couple weeks), so that we won't need to immediately revise the recipe

**Comment by thchang** [comment]
Thanks @dopplershift for your helpful advice and explanation of the process

**Review by dopplershift — APPROVED** [review]

**Pull request merged**
